contrib/inventory/azure_rm.py

```diff
@@ -782,15 +806,30 @@ def _load_machines(self, machines):
 
     def _selected_machines(self, virtual_machines):
         selected_machines = []
+
         for machine in virtual_machines:
             if self._args.host and self._args.host == machine.name:
                 selected_machines.append(machine)
             if self.tags and self._tags_match(machine.tags, self.tags):
                 selected_machines.append(machine)
             if self.locations and machine.location in self.locations:
                 selected_machines.append(machine)
+            if self._args.vmss and self._args.vmss == getattr(machine, 'vmss_name', None):
+                selected_machines.append(machine)
+
         return selected_machines
 
+    def _selected_vmss(self, vmsses):
+        selected_vmsses = []
+
+        if self._args.vmss:
+            for vmss in vmsses:
+                if self._args.vmss == vmss.name:
+                    selected_vmsses.append(vmss)
```

Comment:
why not just "return [vmss]" here? and not continue the loop?

Reply:
fixed